provider/marathon/marathon.go

```diff
@@ -420,6 +447,187 @@ func (p *Provider) hasCircuitBreakerLabels(application marathon.Application) boo
 	return ok
 }
 
+// Map of services properties
+// we can get it with label[serviceName][propertyName] and we got the propertyValue
+type labelServiceProperties map[string]map[string]string
+
+// Check if for the given task and applications, we find labels that are defining services
+func (p *Provider) hasTaskServices(task marathon.Task, applications []marathon.Application) bool {
+	application, err := getApplication(task, applications)
+	if err != nil {
+		log.Errorf("Unable to get marathon application from task %s", task.AppID)
+		return false
+	}
+	return len(extractServicesLabels(application.Labels)) > 0
+}
+
+// Check if for the given application, we find labels that are defining services
+func (p *Provider) hasServices(application marathon.Application) bool {
+	return len(extractServicesLabels(application.Labels)) > 0
+}
+
+// Extract the service labels from application labels of dockerData struct
+func extractServicesLabels(labels *map[string]string) labelServiceProperties {
+	v := make(labelServiceProperties)
+
+	for index, serviceProperty := range *labels {
+		matches := servicesPropertiesRegexp.FindStringSubmatch(index)
+		if matches != nil {
+			result := make(map[string]string)
+			for i, name := range servicesPropertiesRegexp.SubexpNames() {
+				if i != 0 {
+					result[name] = matches[i]
+				}
+			}
+			serviceName := result["service_name"]
+			if _, ok := v[serviceName]; !ok {
+				v[serviceName] = make(map[string]string)
+			}
+			v[serviceName][result["property_name"]] = serviceProperty
+		}
+	}
+
+	return v
```

Comment:
The parser route is definitely possible, though I personally prefer the regex as it will be more performant and less code to maintain.  After the parser logic is all coded up, it will effectively do the same thing as Regex, scan the label text, tokenize it by dot-delimited groups, build the same `result` map with the rest of the logic to follow.

The regex logic is actually not very complex (IMHO).
`^traefik\.(?P<service_name>.*?)\.(?P<property_name>port|portIndex|weight|protocol|frontend\.(.*))$` breaks down as following:
* The label MUST start with `traefik.` (use of `^` indicates start of line)
* The next section, defined by `(?P<service_name>.*?)` group is a named & numbered capturing group (submatch) that will basically extract a piece of text that sits in between `.` separators after `traefik.` segment.
* The next section, defined by `(?P<property_name>port|portIndex|weight|protocol|frontend\.(.*))` group is another named & numbered submatch, but conditional on `property_name` text segment be only one of the listed options: `port`, `portIndex`, `weight`, `protocol` or `frontend.*`
* The rest of the label till end of line/entry, defined by `$`

The reason this regex looks complex is due to using named groups and having to use `\` in many places to escape the `.` :)

The issues with the parser are: 
* instead of 3 lines of code used in regex (regex definition, `servicesPropertiesRegexp.FindStringSubmatch(index)` and `servicesPropertiesRegexp.SubexpNames()`) we would have much more code for the parser logic 
* parser logic would be custom-written, needs to be extensively tested and add complexity that has to be maintained long term
* regex performance is already very good (Google guys have done their part. :) ), the parser might need optimizations to tune it to make sure performance is not suffering.

Thoughts?